lib/src/components/NxCombobox/types.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import * as PropTypes from 'prop-types';
+import { HTMLAttributes, ReactNode } from 'react';
+import DataItem from '../../util/DataItem';
+
+export interface Props<T extends string | number = string>
+  extends Omit<HTMLAttributes<HTMLDivElement>, 'onChange'> {
+  value: string;
+  loading?: boolean | null;
+  loadError?: ReactNode;
+  matches: DataItem<T, string>[];
+  onSearch: (s: string) => void;
+  onChange: (s: string) => void;
+  disabled?: boolean | null;
+  emptyMessage?: ReactNode;
+  autoComplete?: boolean | null;
+  validatable?: boolean | null;
+  isPristine?: boolean | null;
+  trimmedValue?: string | null;
```

Comment:
We don't really want this in the props.  The component must be able to handle it if it's on the props object, but it shouldn't advertise that it can receive it.  See how `NxTextInput` does it.

Reply:
Updated here 7c5c372a68c742ba08985912c1488ddbf608e0d0.